### PR TITLE
feat: dusking dbcs

### PR DESCRIPTION
**Pull request by grumbach**
- includes `sn_dbc` code in `sn_transfers`

gets rid of DBCs and uses the new nomenclature:

- `SNT`: the unit of our currency
- `NanoTokens `: (currently Token) 10^-9 SNT
- `CashNote`: (currently DBC), spendable unique key along with spend related information, containing n nanos. This CashNote contains unique key related information, leaking it damages privacy but it cannot be spent by an attacker since it doesn’t contain your private key.
- `Spend`/`SignedSpend`: Network representation of a spent CashNote, it cannot be linked back to your MainKey. It makes the system auditable and transactions verifiable
- `Transaction`: (currently DbcTransaction), Network operation that creates new CashNote while spending CashNote of the same total value, transaction information is findable both in CashNotes and in the Spends on the Network.
- `CashNoteRedemption`: (currently UTXO) minimal information from which the recipient of a Transaction can redeem their new spendable CashNote
- `Transfer`: encrypted NoteRedemptions ready to be sent to a transaction recipient so they can receive money.

As for keys:

- `MainSecretKey`: (current MainKey) the private main key (secret)
- `DerivedSecretKey`: (current DerivedKey) a derived key from the main key (IS UNIQUE but secret)
- `MainPubkey`: (current PublicAddress) the public key of the main key
- `UniquePubkey`: (current dbc_id) the public key of the derived key (IS UNIQUE)

## Description

reviewpad:summary

### CODE REVIEW

File: sn_cli/src/subcommands/wallet.rs [hunk 1]
```diff
@@ -37,32 +37,32 @@ pub enum WalletCmds {
...
     /// The default received directory is platform specific:
-    ///  - Linux: $HOME/.local/share/safe/wallet/received_dbcs
-    ///  - macOS: $HOME/Library/Application Support/safe/wallet/received_dbcs
-    ///  - Windows: C:\Users\{username}\AppData\Roaming\safe\wallet\received_dbcs
+    ///  - Linux: $HOME/.local/share/safe/wallet/received_cash_notes
+    ///  - macOS: $HOME/Library/Application Support/safe/wallet/received_cash_notes
+    ///  - Windows: C:\Users\{username}\AppData\Roaming\safe\wallet\received_cash_notes
     ///
-    /// If you find the default path unwieldy, you can also set the RECEIVED_DBCS_PATH environment
+    /// If you find the default path unwieldy, you can also set the RECEIVED_CashNoteS_PATH environment
```
Comment (joshuef) [inline sn_cli/src/subcommands/wallet.rs]: `RECEIVED_CashNoteS_PATH` ->  all caps

File: sn_cli/src/subcommands/wallet.rs [hunk 2]
```diff
@@ -200,7 +200,7 @@ fn deposit(root_dir: &Path, read_from_stdin: bool, dbc: Option<String>) -> Resul
     wallet.try_load_deposits()?;
 
     let deposited =
-        sn_dbc::Token::from_nano(wallet.balance().as_nano() - previous_balance.as_nano());
+        sn_transfers::Nano::from_nano(wallet.balance().as_nano() - previous_balance.as_nano());
```
Comment (joshuef) [inline sn_cli/src/subcommands/wallet.rs]: `sn_transfers::Nano::from_nano`

should just be `new` or `from` i think? nano::from_nano is odd

File: sn_client/README.md [hunk 3]
```diff
@@ -81,16 +81,16 @@ let client = Client::new(signer, peers, req_response_timeout, custom_concurrency
...
   - `verify_store: bool`
 - **Returns**: `Result<()>`
 
 #### `get_spend_from_network`
 
-- **Description**: Get a dbc spend from the network.
-- **Parameters**: `dbc_id: &DbcId`
+- **Description**: Get a cash_note spend from the network.
+- **Parameters**: `cash_note_id: &CashNoteId`
 - **Returns**: `Result<SignedSpend>`
```
Comment (grumbach) [inline sn_client/README.md:94]: Docs out of date, should inline them in code `///` and remove this

File: sn_networking/src/driver.rs [hunk 4]
```diff
@@ -60,8 +60,8 @@ type PendingGetRecord = HashMap<QueryId, (oneshot::Sender<Result<Record>>, GetRe
 
 /// What is the largest packet to send over the network.
 /// Records larger than this will be rejected.
-// TODO: revisit once utxo is in
-const MAX_PACKET_SIZE: usize = 1024 * 1024 * 5; // the chunk size is 1mb, so should be higher than that to prevent failures, 5mb here to allow for DBC storage
+// TODO: revisit once cashnote_redemption is in
```
Comment (grumbach) [inline sn_networking/src/driver.rs:63]: Adapt max packet size

**Comment by happybeing** [comment]
Is `CashNote` straying away from utility token and towards money?

**Comment by joshuef** [comment]
> Is CashNote straying away from utility token and towards money?

This change in nomenclature will be elaborated upon a dev update 👍 . 

But to be clear: this PR has no changes to how tokens and transfers operate on the network over the previous `dbc` crate. It's all being rolled in here and naming updated to better reflect what's actually happening

**Review by joshuef — APPROVED** [review]

**Pull request merged**